src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -52,7 +52,7 @@ public abstract partial class JsonConverter
         /// <summary>
         /// Whether the converter is built-in and handles a number type.
         /// </summary>
-        internal bool IsInternalConverterForNumberType;
+        internal virtual bool IsInternalConverterForNumberType { get; }
```

Comment:
I assume this will be inlined and as fast as the field.

Reply:
We no longer use this bool during (de)serialization as shown in this [diff](https://github.com/dotnet/runtime/pull/41679/files#diff-56b52018b267905bd21daf4c7615b6c7L166-L182) and this [diff](https://github.com/dotnet/runtime/pull/41679/files#diff-56b52018b267905bd21daf4c7615b6c7R359); instead we rely only on `state.Current.NumberHandling`. The bool is now only used here during warm up, so I believe a property is good enough: